### ISSUE TITLE
Preset and doc for Mistral (multilingual)

### ISSUE THREAD

**Issue by federicoparra** [keras-team/keras-nlp#1418]
Can't find preset for running Mistral, nor doc about how to.
In addition I see other models such as GPT2 have an English tokenizer preset whereas Mistral is trained to speak perfectly in many languages so the tokenizer should be multilingual not English.

**Comment by tirthasheshpatel** [comment]
> Can't find preset for running Mistral, nor doc about how to.

Yeah, we haven't yet add the presets for Mistral. It will be added by the next release though.



> In addition I see other models such as GPT2 have an English tokenizer preset whereas Mistral is trained to speak perfectly in many languages so the tokenizer should be multilingual not English.

Are the tokenizers for other languages published somewhere openly? The model raw file just contains a Sentence Piece model for english.

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

what is the equivalent for this block in Mistral and Llama.

```
gpt2_preprocessor = keras_nlp.models.GPT2CausalLMPreprocessor.from_preset(
    "gpt2_base_en",
    sequence_length=256,
    add_end_token=True,
)
gpt2_lm = keras_nlp.models.GPT2CausalLM.from_preset("gpt2_base_en", preprocessor=gpt2_preprocessor)
```

Are they developed or not ?

**Comment by tirthasheshpatel** [comment]
> Are they developed or not ?

Not yet, but they will be added by the next release!

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

Thanks for reply .

Happy to hear. 
Shall I know any roadmaps about the next release like tentative dates ?

**Comment by tirthasheshpatel** [comment]
There's no set/tentative release dates as far as I know. Maybe @mattdangerw knows more, but we will probably release the next version late February.

**Comment by RageshAntonyHM** [comment]
@mattdangerw 

Thanks.

I installed keras-nlp from your PR branch 
`!pip install -q git+https://github.com/tirthasheshpatel/keras-nlp.git@mistral-presets`

When executing this block, I get this 403 error:

```
mistrl_preprocessor = MistralPreprocessor.from_preset(
    "mistral_7b_en",
    sequence_length=4096,
    add_end_token=True,
)

gpt2_lm = MistralCausalLM.from_preset("mistral_7b_en", preprocessor=mistrl_preprocessor)
```

```
Downloading from https://www.kaggle.com/api/v1/models/keras/mistral/keras/mistral_7b_en/1/download/tokenizer.json...
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/kagglehub/exceptions.py](https://localhost:8080/#) in kaggle_api_raise_for_status(response, resource_handle)
     48     try:
---> 49         response.raise_for_status()
     50     except requests.HTTPError as e:

10 frames
HTTPError: 403 Client Error: Forbidden for url: https://www.kaggle.com/api/v1/models/keras/mistral/keras/mistral_7b_en/1/download/tokenizer.json

The above exception was the direct cause of the following exception:

KaggleApiHTTPError                        Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/kagglehub/exceptions.py](https://localhost:8080/#) in kaggle_api_raise_for_status(response, resource_handle)
     69 
     70         # Default handling
---> 71         raise KaggleApiHTTPError(message, response=response) from e
     72 
     73 

KaggleApiHTTPError: 403 Client Error.

You don't have permission to access resource at URL: https://www.kaggle.com/models/keras/mistral/frameworks/keras/variations/mistral_7b_en/versions/1
Please make sure you are authenticated if you are trying to access a private resource or a resource requiring consent.
```

I installed kaggle package and also put the kaggle.json.

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel How to fix this permission error above ?

**Comment by tirthasheshpatel** [comment]
We are still testing the model, so weights are private. Once the PR is merged, you should be able to use the model.

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

Yeah got it.

Is it possible to manually provide the model URL either from Kaggle or HuggingFace ?

**Comment by tirthasheshpatel** [comment]
> Is it possible to manually provide the model URL either from Kaggle or HuggingFace ?

You should be able to convert the weights yourself from the huggingface model to the KerasNLP model using [this script](https://github.com/keras-team/keras-nlp/blob/79e38f88fcc697ffce58814a9122c107793d5c69/tools/checkpoint_conversion/convert_mistral_checkpoints.py) from my PR. But that requires having 80-100GB of RAM and I would just recommend waiting till we complete #1436

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

I checked since that PR merged. Now I got different error. I am I doing something wrong ?

```
mistrl_preprocessor = MistralPreprocessor.from_preset(
    "mistral_7b_en",
    sequence_length=4096,
    add_end_token=True,
)

gpt2_lm = MistralCausalLM.from_preset("mistral_7b_en", preprocessor=mistrl_preprocessor)
```

```
Downloading from https://www.kaggle.com/api/v1/models/keras/mistral/keras/mistral_7b_en/3/download/tokenizer.json...
100%|██████████| 325/325 [00:00<00:00, 219kB/s]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[<ipython-input-6-b399292c5eab>](https://localhost:8080/#) in <cell line: 1>()
----> 1 mistrl_preprocessor = MistralPreprocessor.from_preset(
      2     "mistral_7b_en",
      3     sequence_length=4096,
      4     add_end_token=True,
      5 )

4 frames
[/usr/local/lib/python3.10/dist-packages/keras/saving/legacy/serialization.py](https://localhost:8080/#) in class_and_config_for_serialized_keras_object(config, module_objects, custom_objects, printable_module_name)
    366     )
    367     if cls is None:
--> 368         raise ValueError(
    369             f"Unknown {printable_module_name}: '{class_name}'. "
    370             "Please ensure you are using a `keras.utils.custom_object_scope` "

ValueError: Unknown object: 'MistralTokenizer'. Please ensure you are using a `keras.utils.custom_object_scope` and that this object is included in the scope. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
```

**Comment by tirthasheshpatel** [comment]
You should be using `MistralCausalLMPreprocessor` instead of `MistralPreprocessor`:

```python
from keras_nlp.models import MistralCausalLM, MistralCausalLMPreprocessor

preprocessor = MistralCausalLMPreprocessor.from_preset(
    'mistral_7b_en', sequence_length=4096, add_end_token=True
)
generator = MistralCausalLM.from_preset('mistral_7b_en', preprocessor=preprocessor)
```

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

Thanks. But sorry. I get same again

<img width="1057" alt="image" src="https://github.com/keras-team/keras-nlp/assets/141845438/96d25a31-3c47-4290-bd17-6a7625847709">

**Comment by tirthasheshpatel** [comment]
Did you try upgrading Keras? With Keras 3.0.4 and 3.0.5, I am able to use the model without errors.

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

I installed new keras. These are my colab steps

`!pip install --upgrade keras
`

`!pip install git+https://github.com/keras-team/keras-nlp.git tensorflow-text==2.12`

```
import numpy as np
import keras_nlp
from keras_nlp.backend import ops
from keras_nlp.models import MistralCausalLM, MistralCausalLMPreprocessor
from keras_nlp.models.mistral.mistral_backbone import MistralBackbone
from keras_nlp.models.mistral.mistral_causal_lm import MistralCausalLM
from keras_nlp.models.mistral.mistral_tokenizer import MistralTokenizer
from keras_nlp.models import MistralCausalLM, MistralCausalLMPreprocessor

preprocessor = MistralCausalLMPreprocessor.from_preset(
    'mistral_7b_en', sequence_length=4096, add_end_token=True
)
generator = MistralCausalLM.from_preset('mistral_7b_en', preprocessor=preprocessor)
```

But same error again

```
Downloading from https://www.kaggle.com/api/v1/models/keras/mistral/keras/mistral_7b_en/3/download/tokenizer.json...
100%|██████████| 325/325 [00:00<00:00, 199kB/s]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[<ipython-input-2-a6761ab59ed2>](https://localhost:8080/#) in <cell line: 1>()
----> 1 preprocessor = MistralCausalLMPreprocessor.from_preset(
      2     'mistral_7b_en', sequence_length=4096, add_end_token=True
      3 )
      4 generator = MistralCausalLM.from_preset('mistral_7b_en', preprocessor=preprocessor)

5 frames
[/usr/local/lib/python3.10/dist-packages/keras/saving/legacy/serialization.py](https://localhost:8080/#) in class_and_config_for_serialized_keras_object(config, module_objects, custom_objects, printable_module_name)
    366     )
    367     if cls is None:
--> 368         raise ValueError(
    369             f"Unknown {printable_module_name}: '{class_name}'. "
    370             "Please ensure you are using a `keras.utils.custom_object_scope` "

ValueError: Unknown object: 'MistralTokenizer'. Please ensure you are using a `keras.utils.custom_object_scope` and that this object is included in the scope. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
```

**Comment by tirthasheshpatel** [comment]
KerasNLP overrides the installation of Keras with an older version. You can move the `!pip install git+https://github.com/keras-team/keras-nlp.git tensorflow-text==2.12` above `!pip install --upgrade keras` and everything should work as expected.

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

Yeah. You are correct. That git clone install installs Keras 2.x !!. After I moved the order like you told, It started to download. 

After that I got this error [ResourceExhaustedError] :

```
Downloading from https://www.kaggle.com/api/v1/models/keras/mistral/keras/mistral_7b_en/3/download/config.json...
100%|██████████| 443/443 [00:00<00:00, 943kB/s]

ResourceExhaustedError                    Traceback (most recent call last)
[<ipython-input-5-a6761ab59ed2>](https://localhost:8080/#) in <cell line: 4>()
      2     'mistral_7b_en', sequence_length=4096, add_end_token=True
      3 )
----> 4 generator = MistralCausalLM.from_preset('mistral_7b_en', preprocessor=preprocessor)

8 frames
[/usr/local/lib/python3.10/dist-packages/keras_nlp/models/mistral/mistral_transformer_decoder.py](https://localhost:8080/#) in build(self, decoder_sequence_shape)
     99             name="feedforward_intermediate_dense",
    100         )
--> 101         self._feedforward_intermediate_dense.build(decoder_sequence_shape)
    102 
    103         self._feedforward_gate_dense = keras.layers.Dense(

ResourceExhaustedError: {{function_node __wrapped__AddV2_device_/job:localhost/replica:0/task:0/device:GPU:0}} OOM when allocating tensor with shape[4096,14336] and type bfloat16 on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc [Op:AddV2] name:
```

Is it due to GPU VRAM (colab's free T4) issue or some thing ?

**Comment by tirthasheshpatel** [comment]
Yup, you will need at least 16GB VRAM to load the model (I don't think free tier of Colab offers that much RAM though).

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

Yeah. GPU alloc shows  13.7 / 15.0 GB

Okay. Let me try Runpod or other things with 24GB cards.

Thanks very much for your support here

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 

I tried running the entire code block

```
preprocessor = MistralCausalLMPreprocessor.from_preset(
    'mistral_7b_en', sequence_length=4096, add_end_token=True
)
generator = MistralCausalLM.from_preset('mistral_7b_en', preprocessor=preprocessor)

def run_inference(input, generate_tflite):
  interp = interpreter.InterpreterWithCustomOps(
      model_content=generate_tflite,
      custom_op_registerers=tf_text.tflite_registrar.SELECT_TFTEXT_OPS)
  interp.get_signature_list()

  generator = interp.get_signature_runner('serving_default')
  output = generator(prompt=np.array([input]))
  print("\nGenerated with TFLite:\n", output["output_0"])

@tf.function
def generate(prompt, max_length):
    return generator.generate(prompt, max_length)

concrete_func = generate.get_concrete_function(tf.TensorSpec([], tf.string), 100)


generator.jit_compile = False
converter = tf.lite.TFLiteConverter.from_concrete_functions([concrete_func],
                                                            generator)
converter.target_spec.supported_ops = [
  tf.lite.OpsSet.TFLITE_BUILTINS, # enable TensorFlow Lite ops.
  tf.lite.OpsSet.SELECT_TF_OPS # enable TensorFlow ops.
]
converter.allow_custom_ops = True
converter.target_spec.experimental_select_user_tf_ops = ["UnsortedSegmentJoin", "UpperBound"]
converter._experimental_guarantee_all_funcs_one_use = True
generate_tflite = converter.convert()
run_inference("I'm enjoying a", generate_tflite)

with open('unquantized_mistral.tflite', 'wb') as f:
  f.write(generate_tflite)
```

**I got this error**
```
   AttributeError: in user code:
at line33: generator.generate(prompt, max_length)
........
    File "/usr/local/lib/python3.10/dist-packages/keras_nlp/utils/tensor_utils.py", line 48, in tensor_to_list  *
        list_outputs = inputs.numpy()

    AttributeError: 'SymbolicTensor' object has no attribute 'numpy'
```

I searched the internet and found a similar issue https://github.com/keras-team/keras-nlp/issues/1090 

Do I need to wait until full release after 0.7.0 that is still tasks are pending ? OR any fix exists ?

**Comment by tirthasheshpatel** [comment]
You should see the example in #998. The `generator.generate` method is not JITable. So, you will have to use the `generate_function` instead which accets a dictionary of token_ids and padding_mask generated by the tokenizer. You will then need to run the tokenizer outside the TF graph. Something like this:

```
preprocessor = MistralCausalLMPreprocessor.from_preset(
    'mistral_7b_en', sequence_length=4096, add_end_token=True
)
generator = MistralCausalLM.from_preset('mistral_7b_en', preprocessor=preprocessor)

def run_inference(input, generate_tflite):
  interp = interpreter.InterpreterWithCustomOps(
      model_content=generate_tflite,
      custom_op_registerers=tf_text.tflite_registrar.SELECT_TFTEXT_OPS)
  interp.get_signature_list()

  preprocessor_output = preprocessor.generate_preprocess(
    input, sequence_length=preprocessor.sequence_length
  )
  generator = interp.get_signature_runner('serving_default')
  output = generator(preprocessor_output)
  output = preprocessor.generate_postprocess(output["output_0"])
  print("\nGenerated with TFLite:\n", output)

generate_function = generator.make_generate_function()
concrete_func = generate_function.get_concrete_function({
  "token_ids": tf.TensorSpec([None, 100]),
  "padding_mask": tf.TensorSpec([None, 100])
}, end_token_id=2)


converter = tf.lite.TFLiteConverter.from_concrete_functions([concrete_func],
                                                            generator)
converter.target_spec.supported_ops = [
  tf.lite.OpsSet.TFLITE_BUILTINS, # enable TensorFlow Lite ops.
  tf.lite.OpsSet.SELECT_TF_OPS # enable TensorFlow ops.
]
converter.allow_custom_ops = True
converter.target_spec.experimental_select_user_tf_ops = ["UnsortedSegmentJoin", "UpperBound"]
converter._experimental_guarantee_all_funcs_one_use = True
generate_tflite = converter.convert()
run_inference("I'm enjoying a", generate_tflite)

with open('unquantized_mistral.tflite', 'wb') as f:
  f.write(generate_tflite)
```

**Comment by RageshAntonyHM** [comment]
@tirthasheshpatel 
Thanks. That error resolved but I got an error: 

`ValueError: Message tensorflow.GraphDef exceeds maximum protobuf size of 2GB: 14487363127
`

```
loading generate_tflite
2024-02-16 04:21:19.978439: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 1
2024-02-16 04:21:19.978643: I tensorflow/core/grappler/clusters/single_machine.cc:361] Starting new session
2024-02-16 04:21:19.980975: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1929] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 22503 MB memory:  -> device: 0, name: NVIDIA RTX A5000, pci bus id: 0000:a1:00.0, compute capability: 8.6
Traceback (most recent call last):
  File "/workspace/mistral.py", line 62, in <module>
    generate_tflite = converter.convert()
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 2185, in convert
    return super(TFLiteConverterV2, self).convert()
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1139, in wrapper
    return self._convert_and_export_metrics(convert_func, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1093, in _convert_and_export_metrics
    result = convert_func(self, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1785, in convert
    self._freeze_concrete_function()
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert_phase.py", line 215, in wrapper
    raise error from None  # Re-throws the exception.
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert_phase.py", line 205, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1666, in _freeze_concrete_function
    _convert_to_constants.convert_variables_to_constants_v2_as_graph(
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/convert_to_constants.py", line 1253, in convert_variables_to_constants_v2_as_graph
    frozen_func = _construct_concrete_function(func, output_graph_def,
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/convert_to_constants.py", line 1097, in _construct_concrete_function
    new_func = wrap_function.function_from_graph_def(output_graph_def,
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 674, in function_from_graph_def
    wrapped_import = wrap_function(_imports_graph_def, [])
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 638, in wrap_function
    func_graph.func_graph_from_py_func(
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/func_graph.py", line 1059, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 85, in __call__
    return self.call_with_variable_creator_scope(self._fn)(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 91, in wrapped
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 668, in _imports_graph_def
    importer.import_graph_def(graph_def, name="")
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/util/deprecation.py", line 588, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/importer.py", line 407, in import_graph_def
    return _import_graph_def_internal(
  File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/importer.py", line 508, in _import_graph_def_internal
    with c_api_util.tf_buffer(graph_def.SerializeToString()) as serialized:
ValueError: Message tensorflow.GraphDef exceeds maximum protobuf size of 2GB: 14487363127
```

Any configuration needed ?

**Comment by federicoparra** [comment]
> @tirthasheshpatel Thanks. I ran the code you provided. That error resolved but I got an error:
> 
> `ValueError: Message tensorflow.GraphDef exceeds maximum protobuf size of 2GB: 14487363127 `
> 
> ```
> running converter.convert() for generate_tflite
> 
> 2024-02-16 04:21:19.978439: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 1
> 2024-02-16 04:21:19.978643: I tensorflow/core/grappler/clusters/single_machine.cc:361] Starting new session
> 2024-02-16 04:21:19.980975: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1929] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 22503 MB memory:  -> device: 0, name: NVIDIA RTX A5000, pci bus id: 0000:a1:00.0, compute capability: 8.6
> Traceback (most recent call last):
>   File "/workspace/mistral.py", line 62, in <module>
>     generate_tflite = converter.convert()
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 2185, in convert
>     return super(TFLiteConverterV2, self).convert()
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1139, in wrapper
>     return self._convert_and_export_metrics(convert_func, *args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1093, in _convert_and_export_metrics
>     result = convert_func(self, *args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1785, in convert
>     self._freeze_concrete_function()
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert_phase.py", line 215, in wrapper
>     raise error from None  # Re-throws the exception.
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/convert_phase.py", line 205, in wrapper
>     return func(*args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/lite/python/lite.py", line 1666, in _freeze_concrete_function
>     _convert_to_constants.convert_variables_to_constants_v2_as_graph(
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/convert_to_constants.py", line 1253, in convert_variables_to_constants_v2_as_graph
>     frozen_func = _construct_concrete_function(func, output_graph_def,
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/convert_to_constants.py", line 1097, in _construct_concrete_function
>     new_func = wrap_function.function_from_graph_def(output_graph_def,
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 674, in function_from_graph_def
>     wrapped_import = wrap_function(_imports_graph_def, [])
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 638, in wrap_function
>     func_graph.func_graph_from_py_func(
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/func_graph.py", line 1059, in func_graph_from_py_func
>     func_outputs = python_func(*func_args, **func_kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 85, in __call__
>     return self.call_with_variable_creator_scope(self._fn)(*args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 91, in wrapped
>     return fn(*args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/eager/wrap_function.py", line 668, in _imports_graph_def
>     importer.import_graph_def(graph_def, name="")
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/util/deprecation.py", line 588, in new_func
>     return func(*args, **kwargs)
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/importer.py", line 407, in import_graph_def
>     return _import_graph_def_internal(
>   File "/usr/local/lib/python3.10/dist-packages/tensorflow/python/framework/importer.py", line 508, in _import_graph_def_internal
>     with c_api_util.tf_buffer(graph_def.SerializeToString()) as serialized:
> ValueError: Message tensorflow.GraphDef exceeds maximum protobuf size of 2GB: 14487363127
> ```
> 
> Any configuration needed ? I have tried lot of techniques from internet, nothing works

I have this very same issue :( did you find an answer?

**Comment by RageshAntonyHM** [comment]
@federicoparra 
No. I have tried lot of other ways. Everything ends with same error above. For example, I tried to generate  tflite mode via ONNX model that converted from transformers model (Transformers => ONNX ) using Optimum and (ONNX => TFlite) using onnx-tensorflow library. 

But even in ONNX => TFlite phase, I got the same `protobuf size of 2GB` issue. 

I doubt exporting models above 2GB not possible for now due to protobuf limitations at the Tensorflow core itself 

waiting for @tirthasheshpatel @mattdangerw responses.
